services/provider/server/server.go

```diff
@@ -1410,29 +1410,26 @@ func (s *OCSProviderServer) appendCephClientSecretKubeResources(
 	consumerConfig util.StorageConsumerResources,
 ) ([]client.Object, error) {
 
-	cephClients := []string{}
-	if consumerConfig.GetCsiRbdProvisionerCephUserName() != "" {
-		cephClients = append(cephClients, consumerConfig.GetCsiRbdProvisionerCephUserName())
-	}
-	if consumerConfig.GetCsiRbdNodeCephUserName() != "" {
-		cephClients = append(cephClients, consumerConfig.GetCsiRbdNodeCephUserName())
-	}
-	if consumerConfig.GetCsiCephFsProvisionerCephUserName() != "" {
-		cephClients = append(cephClients, consumerConfig.GetCsiCephFsProvisionerCephUserName())
-	}
-	if consumerConfig.GetCsiCephFsNodeCephUserName() != "" {
-		cephClients = append(cephClients, consumerConfig.GetCsiCephFsNodeCephUserName())
-	}
-	if consumerConfig.GetCsiNfsProvisionerCephUserName() != "" {
-		cephClients = append(cephClients, consumerConfig.GetCsiNfsProvisionerCephUserName())
+	csiCephUserGeneration, err := util.GetCsiCephUserCurrGeneration()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get csi ceph user generation: %v", err)
 	}
-	if consumerConfig.GetCsiNfsNodeCephUserName() != "" {
-		cephClients = append(cephClients, consumerConfig.GetCsiNfsNodeCephUserName())
+
+	cephClientsToSecrets := map[string]string{
+		util.GenerateCsiRbdProvisionerCephClientName(csiCephUserGeneration, consumer.UID):    consumerConfig.GetCsiRbdProvisionerCephUserName(),
+		util.GenerateCsiRbdNodeCephClientName(csiCephUserGeneration, consumer.UID):           consumerConfig.GetCsiRbdNodeCephUserName(),
+		util.GenerateCsiCephFsProvisionerCephClientName(csiCephUserGeneration, consumer.UID): consumerConfig.GetCsiCephFsProvisionerCephUserName(),
+		util.GenerateCsiCephFsNodeCephClientName(csiCephUserGeneration, consumer.UID):        consumerConfig.GetCsiCephFsNodeCephUserName(),
+		util.GenerateCsiNfsProvisionerCephClientName(csiCephUserGeneration, consumer.UID):    consumerConfig.GetCsiNfsProvisionerCephUserName(),
+		util.GenerateCsiNfsNodeCephClientName(csiCephUserGeneration, consumer.UID):           consumerConfig.GetCsiNfsNodeCephUserName(),
 	}
 
-	for i := range cephClients {
+	for cephClientName, secretName := range cephClientsToSecrets {
+		if secretName == "" {
+			continue
+		}
 		cephClient := &rookCephv1.CephClient{}
-		cephClient.Name = cephClients[i]
+		cephClient.Name = cephClientName
```

Comment:
Rook is generating the secret with the same name that we have for the cephClients. Do we need to get the ceph clients to get the secret?